daffodil-core/src/main/scala/org/apache/daffodil/dpath/Expression.scala

```diff
@@ -1553,46 +1556,91 @@ case class FunctionCallExpression(functionQNameString: String, expressions: List
         FNTwoArgsExprInferedArgType(functionQNameString, functionQName, args,
-          NodeInfo.String, NodeInfo.String, args(1).inherentType, DFDLXOutputTypeCalcString(_))
-
-      case (RefQName(_, "outputTypeCalcNextSiblingInt", DFDLX), args) =>
-        FNZeroArgExpr(functionQNameString, functionQName,
-          NodeInfo.Int, NodeInfo.AnyAtomic, DFDLXOutputTypeCalcNextSiblingInt(_, _))
-
-      case (RefQName(_, "outputTypeCalcNextSiblingString", DFDLX), args) =>
-        FNZeroArgExpr(functionQNameString, functionQName,
-          NodeInfo.String, NodeInfo.AnyAtomic, DFDLXOutputTypeCalcNextSiblingString(_, _))
+          typeCalc.srcType, NodeInfo.String, typeCalc.dstType, DFDLXOutputTypeCalc(_, typeCalc))
+      }
 
-      case (RefQName(_, "repTypeValueInt", DFDLX), args) =>
+      case (RefQName(_, "outputTypeCalcNextSibling", DFDLX), args) => {
+        val erd = compileInfo.lexicalContextRuntimeData match{
+          case erd: ElementRuntimeData => erd
+          case _ => SDE("dfdlx:outputTypeCalcNextSibling can only be defined on an element")
+        }
+        val resolver = erd.nextElementResolver
+        //we keep the ERD to be able to produce better error messages
+        val dstTypes:Seq[(NodeInfo.Kind, ElementRuntimeData)] = resolver.allPossibleNextElements.map(erd => {
+          val strd = erd.optSimpleTypeRuntimeData match{
+            case Some(x) => x
+            case None => SDE("dfdlx:outputTypeCalcNextSibling: potential next sibling %s does not have a simple type def. This could be because it has a primitive type, or a complex type.", erd.namedQName)
+          }
+          val calc = strd.typeCalculator match{
+            case Some(x) => x
+            case None => SDE("dfdlx:outputTypeCalcNextSibling(): potential next sibling %s does not define a type calculator", erd.namedQName)
+          }
+          if(!calc.supportsUnparse){
+            SDE("dfdlx:outputTypeCalcNextSibling(): potential next sibling %s does not have an outputTypeCalc", erd.namedQName)
+          }
+          (calc.srcType, erd)
+        })
+        val dstType = dstTypes match{
+          case Seq() => SDE("dfdlx:outputTypeCalcNextSibling() called where no next sibling exists")
+          case Seq(x) => x._1
+          case x +: xs => {
+            val (ans, headQName) = x
+            /*
+             * Verify that all next siblings have the same dstType
+             * In theory, we could loosen the requirement to having the same primitive type,
+             * but in the interest of being conservative, we will stick to the stricter interperatation for now
+             */
+            xs.map(other => {
+              val (otherAns, otherQName) = other
+              if(otherAns != ans){
+                SDE("dfdlx:outputTypeCalcNextSibling() requires that all the possible next siblings have the same "+
+                     "repType. However, the potential next siblings %s and %s have repTypes %s and %s respectivly",
```

Comment:
typo "respectively"

I found the corresponding test that exercises this error, and that is good to go as it doesn't test for this word.